### PR TITLE
chore!: simplify peer id parsing

### PR DESCRIPTION
**Pull request by mversic**
* simplify parsing of peer ids (serialize/deserialize it from string now)
`ed01204EE2FCD53E1730AF142D1E23951198678295047F9314B4006B0CB61850B1DB10@irohad2:1339` instead of:
`{
  "address":"irohad2:1339",
  "public_key":"ed01204EE2FCD53E1730AF142D1E23951198678295047F9314B4006B0CB61850B1DB10"
  }`
* also move `[[sumeragi.trusted_peers]]` table into `trusted_peers` array in the toml config
this is a required field so it should stand out of config subsections

# Migration guide

This change will be reflected in `peer_config.toml` file.
Minimal configuration before:
```
chain = "00000000-0000-0000-0000-000000000000"
public_key = "ed01201C61FAF8FE94E253B93114240394F79A607B7FA55F9E5A41EBEC74B88055768B"
private_key = "802620282ED9F3CF92811C3818DBC4AE594ED59DC1A2F78E4241E31924E101D6B1FB83"

[[sumeragi.trusted_peers]]                                                                 
address = "127.0.0.1:1337"                                                                 
public_key = "ed01201C61FAF8FE94E253B93114240394F79A607B7FA55F9E5A41EBEC74B88055768B"      
                                                                                           
[[sumeragi.trusted_peers]]                                                                 
address = "127.0.0.1:1338"                                                                 
public_key = "ed0120CC25624D62896D3A0BFD8940F928DC2ABF27CC57CEFEB442AA96D9081AAE58A1"      
                                                                                           
[[sumeragi.trusted_peers]]                                                                 
address = "127.0.0.1:1339"                                                                 
public_key = "ed0120FACA9E8AA83225CB4D16D67F27DD4F93FC30FFA11ADC1F5C88FD5495ECC91020"      
                                                                                           
[[sumeragi.trusted_peers]]                                                                 
address = "127.0.0.1:1340"                                                                 
public_key = "ed01208E351A70B6A603ED285D666B8D689B680865913BA03CE29FB7D13A166C4E7F1F"
```
Minimal configuration now:
```toml
chain = "00000000-0000-0000-0000-000000000000"
public_key = "ed01201C61FAF8FE94E253B93114240394F79A607B7FA55F9E5A41EBEC74B88055768B"
private_key = "802620282ED9F3CF92811C3818DBC4AE594ED59DC1A2F78E4241E31924E101D6B1FB83"

trusted_peers = [
  "ed01201C61FAF8FE94E253B93114240394F79A607B7FA55F9E5A41EBEC74B88055768B@127.0.0.1:1337",
  "ed0120CC25624D62896D3A0BFD8940F928DC2ABF27CC57CEFEB442AA96D9081AAE58A1@127.0.0.1:1338",
  "ed0120FACA9E8AA83225CB4D16D67F27DD4F93FC30FFA11ADC1F5C88FD5495ECC91020@127.0.0.1:1339",
  "ed01208E351A70B6A603ED285D666B8D689B680865913BA03CE29FB7D13A166C4E7F1F@127.0.0.1:1340",
]
```

### CODE REVIEW

**Review by 0x009922 — CHANGES_REQUESTED** [review]
When will we stop breaking configuration... @BAStos525 please take a note.

**Comment by 0x009922** [comment]
Also, please, fill the PR description appropriately, at least the migration guide. It directly affects public API.

File: defaults/docker-compose.local.yml [hunk 1]
```diff
@@ -118,7 +131,10 @@ services:
       P2P_ADDRESS: 0.0.0.0:1340
       API_ADDRESS: 0.0.0.0:8083
       GENESIS_PUBLIC_KEY: ed01204164BF554923ECE1FD412D241036D863A6AE430476C898248B8237D77534CFC4
-      TRUSTED_PEERS: '[{"address":"irohad2:1339","public_key":"ed01204EE2FCD53E1730AF142D1E23951198678295047F9314B4006B0CB61850B1DB10"},{"address":"irohad1:1338","public_key":"ed01209897952D14BDFAEA780087C38FF3EB800CB20B882748FC95A575ADB9CD2CB21D"},{"address":"irohad0:1337","public_key":"ed0120A98BAFB0663CE08D75EBD506FEC38A84E576A7C9B0897693ED4B04FD9EF2D18D"}]'
+      TRUSTED_PEERS:
+      - ed01204EE2FCD53E1730AF142D1E23951198678295047F9314B4006B0CB61850B1DB10@irohad2:1339
+      - ed01209897952D14BDFAEA780087C38FF3EB800CB20B882748FC95A575ADB9CD2CB21D@irohad1:1338
+      - ed0120A98BAFB0663CE08D75EBD506FEC38A84E576A7C9B0897693ED4B04FD9EF2D18D@irohad0:1337
```
[the remark this answers is not in the document]
Comment (mversic) [reply defaults/docker-compose.local.yml]: you are right, this was all bs. I started with just wanting to simplify peer parsing. I reverted all this. I apologize.

File: crates/iroha_test_network/src/lib.rs
Comment (s8sato) [inline crates/iroha_test_network/src/lib.rs:1]: Some integration tests are failing due to configuration errors, but otherwise looks good to me

Comment (mversic) [reply crates/iroha_test_network/src/lib.rs:1]: I fixed it now

**Review by s8sato — APPROVED** [review]

**Review by 0x009922 — APPROVED** [review]

**Pull request merged**